PotionEffectUtils - fix null in docs

ShaneBeee opened the pull request:
### Description
This PR aims to fix "null" showing up in the docs for PotionEffectType:
<img width="461" alt="Screenshot 2025-01-02 at 12 00 56 PM" src="https://github.com/user-attachments/assets/7c2172f3-cb91-4246-bd28-2ec5746dd98d" />


---
**Target Minecraft Versions:** any <!-- 'any' means all supported versions -->
**Requirements:** none <!-- Required plugins, server software... -->
**Related Issues:** none <!-- Links to related issues -->

Moderocky commented:
Do you know if there is any way of removing the getId entirely? It's deprecated for removal (though that's hidden by the suppression).

ShaneBeee commented:
> Do you know if there is any way of removing the getId entirely? It's deprecated for removal (though that's hidden by the suppression).

@Moderocky ok I re-wrote the whole PR.
- changed PotionEffectUtils to use keys instead of deprecated IDs
- Added a PotionEffectType registry (when available)

Moderocky reviewed (approved):
Looks good, thanks for fixing this.

Fusezion commented on src/main/java/ch/njol/skript/classes/data/BukkitClasses.java:
```diff
@@ -1015,73 +1015,19 @@ protected boolean canBeInstantiated() {
 
-		Classes.registerClass(new ClassInfo<>(PotionEffectType.class, "potioneffecttype")
-				.user("potion( ?effect)? ?types?") // "type" had to be made non-optional to prevent clashing with potion effects
+		Registry<PotionEffectType> petRegistry = BukkitUtils.getPotionEffectTypeRegistry();
+		if (petRegistry != null) {
```
Everything seems fine, just concerned about `petRegistry` being misleading

ShaneBeee replied:
[P]otion[E]ffect[T]ype... and it's not used outside this class.
I just wanted to keep it short.
Really no reason for it to be anything else.

APickledWalrus commented on src/main/java/ch/njol/skript/util/PotionEffectUtils.java:
```diff
@@ -341,4 +334,74 @@ public static List<PotionEffect> getEffects(ItemType itemType) {
+				public boolean canBeInstantiated() {
+					return false;
+				}
+
+				@Override
+				public void deserialize(final PotionEffectType o, final Fields f) {
+					assert false;
+				}
```
```suggestion
```
Can be removed while we're here

ShaneBeee replied:
When I remove it, it tells me I need it.

APickledWalrus replied:
Maybe it's using a different serializer class. Not a huge deal though, we can ignore this.

APickledWalrus commented on src/main/java/ch/njol/skript/util/PotionEffectUtils.java:
```diff
@@ -341,4 +334,74 @@ public static List<PotionEffect> getEffects(ItemType itemType) {
 		return effects;
 	}
 
+	public static ClassInfo<PotionEffectType> getLegacyClassInfo() {
```
It might be worth annotating/documenting this as Internal (see `ApiStatus.Internal` annotation)

APickledWalrus commented on src/main/java/ch/njol/skript/bukkitutil/BukkitUtils.java:
```diff
@@ -18,4 +19,20 @@ public static boolean registryExists(String registry) {
+		} else if (registryExists("EFFECT")) { // Bukkit (1.21.x)
+			return Registry.EFFECT;
+		}
+		return null;
```
It might be better to throw an exception here (maybe like IllegalState or UnsupportedOperation)

ShaneBeee replied:
I use the nullness in the BukkitClasses class:
<img width="904" alt="Screenshot 2025-01-03 at 10 51 07 AM" src="https://github.com/user-attachments/assets/a4a4604e-7592-49d7-a7fe-98bde013fde8" />

APickledWalrus replied:
I would annotate the method as Nullable then

APickledWalrus reviewed (changes requested):
A few minor things. One thing I wanted to ask about is the serialization changes. Is it possible that the serialization/deserialization process would change on the same version (that is, using the registry vs using PotionEffectType.getByName). Nice work!

ShaneBeee commented:
> Is it possible that the serialization/deserialization process would change on the same version (that is, using the registry vs using PotionEffectType.getByName). Nice work!

It shouldn't. The method just uses Keys internally anyways:
(from bukkit)
<img width="904" alt="Screenshot 2025-01-03 at 10 47 32 AM" src="https://github.com/user-attachments/assets/be594b01-79a0-4a27-9dcd-6d41d2cd85bf" />

APickledWalrus commented on src/main/java/ch/njol/skript/util/PotionEffectUtils.java:
```diff
@@ -341,4 +335,79 @@ public static List<PotionEffect> getEffects(ItemType itemType) {
 		return effects;
 	}
 
+	/** Legacy class info for PotionEffectType (pre-registry)
+	 *
```
```suggestion
	/**
	 * Legacy class info for PotionEffectType (pre-registry)
```

ShaneBeee replied:
OOpsies
